cse_hub/homepage/forms.py

```diff
@@ -0,0 +1,28 @@
+from django import forms
+from django.contrib.auth.models import User
+from django.contrib.auth.forms import UserCreationForm
+
+class RegistrationForm(UserCreationForm):
+    email = forms.EmailField(required = True)
+
+    class Meta:
+        model = User
+        fields = (
+            'username',
+            'first_name',
+            'last_name',
+            'email',
+            'password1',
+            'password2'
+        )
+
+    def save(self, commit=True):
+        user = super(RegistrationForm, self).save(commit=False)
+        user.first_name = self.cleaned_data['first_name']
+        user.last_name = self.cleaned_data['last_name']
+        user.email = self.cleaned_data['email']
+
+        if commit:
```

Comment:
- Remove ```if commit```, as user is not created. Perhaps, due to commit=False in user=super(..) line
- All work is done using Profile model (users.models.py) and not using the inbuilt User model (e.g. number of problems solved, submission files etc). and having a user without a profile might break code.
        So, when a user is created, a corresponding Profile should also be created. Try to login using newly registered user and try submitting some solutions and going back to the profile page (This pr might have made you realise the use of good unit test)
- Wil need to add user profile option as well (perhaps a file field in Profile model, with default value set to some image), but let that be for some other pull request/ issue.